Feature/add new UI steps

StRhys opened the pull request:
### What

Adds a number of new ui steps for general usage.

***Notes***
Not for here but maybe a later date
Do we need `theBetaBannerShouldBeVisible` and `theImproveThisPageBannerShouldBeVisible`, they both use just use `ElementShouldBeVisible`.  Might be missing the generalisation of this library.

On `innerListElementsShouldHaveText`, I might be misunderstanding what this is supposed to be doing but if we add the below html to `main.go` and run the the steps with these steps: 
```
        And the page should contain "breadcrumbs" with list element text "Home,Areas" at 3 depth
        And the page should contain "breadcrumbs" with list element text "Home,Areas" at 2 depth
        And the page should contain "breadcrumbs" with list element text "Home,Areas" at 4 depth
        And the page should contain "breadcrumbs" with list element text "Home,Areas,Third" at 3 depth
        And the page should contain "breadcrumbs" with list element text "Home,Second" at 3 depth
```
I would expect only the first one to pass, but when running `make test-component` it passes them all.
This is based on the one place I could find it being used [here](https://github.com/ONSdigital/dp-frontend-area-profiles/blob/d7744d3953ae45e4abc015bd17fbaa6300cdf189/features/area_page.feature#L29).
```
<div data-test="breadcrumbs">
  <div class="ons-page__container ons-container">
    <nav
      class="ons-breadcrumb print--hide"
    >
      <ol class="ons-breadcrumb__items ons-u-fs-s">
        <li class="ons-breadcrumb__item ons-u-p-no">
	  <a
            class="ons-breadcrumb__link"
            href="/"
	  >Home</a>
        </li>
        <li class="ons-breadcrumb__item ons-u-p-no">
	  <a
            class="ons-breadcrumb__link"
            href="/"
	  >Areas</a>
        </li>
      </ol>
    </nav>
  </div>
</div>
```

It's also causing a date race issue: `race detected during execution of test`. I think the issue is with `RequestChildNodes`, so this might need a whole rewrite or removal.

### How to review

Sense check.
Tests pass.

### Who can review

Not me

lindenmckenzie commented on examples/ui/features/example.feature:
```diff
@@ -1,9 +1,16 @@
 
     Scenario: View web page scenario
-        When I navigate to "/example"
+        Given I navigate to "/example"
```
Given is usually for setup steps.

lindenmckenzie reviewed (changes requested):
Few changes to increase readability for the future. 

re: the other notes - that breadcrumb one seems very obfuscated. If we're not using it anywhere, let's bin it. The banner ones - I don't mind if we have unified UIs having specific callouts (e.g. nav, footer etc. is useful to prevent re-writing code everywhere) but this seems excessive. I would also check that I don't think we're using them both. There is also sometimes a case to be made that sometimes we use a different function for BDD purposes even if the underlying check is the same.

StRhys commented:
Removed the breadcrumbs one, was only being used on the [dp-frontend-area-profiles](https://github.com/ONSdigital/dp-frontend-area-profiles/blob/d7744d3953ae45e4abc015bd17fbaa6300cdf189/features/area_page.feature), looks like this was to be decommissioned at some point.

The beta phase banner is also used on the area profiles, along with [dp-hello-world](https://github.com/ONSdigital/dp-cli/blob/3408f0d575b0e61c339f73efdccd4cd84e5ce9ce/project_generation/content/templates/controller/features/hello_world.feature.tmpl#L4) and [dp-cli](https://github.com/ONSdigital/dp-cli/blob/3408f0d575b0e61c339f73efdccd4cd84e5ce9ce/project_generation/content/templates/controller/features/hello_world.feature.tmpl#L4), which is `hello_world.feature.tmpl`. So have removed it.

The same with improve this banner but that's also being used on the dp-frontend-homepage-controller, [here](https://github.com/ONSdigital/dp-frontend-homepage-controller/blob/65981c468a1cdc48abb426747e452e14138de7de/features/census/census_page.feature#L4) and [here](https://github.com/ONSdigital/dp-frontend-homepage-controller/blob/65981c468a1cdc48abb426747e452e14138de7de/features/homepage/home_page.feature#L4). Also removed this.

Will need to go through and update these repos.
